### ISSUE TITLE
Null fields aren't included on `JsonInclude.Include.ALWAYS`

### ISSUE THREAD

**Issue by toadzky** [dropwizard/dropwizard#1482]
I'm trying to force null fields to be written. When I tried it with a newly constructed object mapper, it works fine. But the object mapper used in the message body writer doesn't write them out.

I'm configuring the ObjectMapper in the run method of my application subclass: 

```
env.getObjectMapper().setSerializationInclusion(JsonInclude.Include.ALWAYS);
```

**Comment by nickbabcock** [comment]
Can you provide a reproducible test? I tested the object mapper created by dropwizard and it was able to serialize null fields to `null` without setting `JsonInclude.Include.ALWAYS`. I also confirmed it working in `ResourceTestRule`

**Comment by toadzky** [comment]
I went looking for the code that was having a problem and it's been changed. So I wrote a quick project to demonstrate the problem.
[dropwizard-serialization-test.zip](https://github.com/dropwizard/dropwizard/files/195926/dropwizard-serialization-test.zip)

 Here's the data classes:
```java
@Value
public class InnerDataClass {
    private int x, y;
}
```
```java
@JsonInclude(ALWAYS)
@Value
public class DataClass {
    private String field1;
    private InnerDataClass field2;
}
```

with a couple of simple `GET`s:
```java
@GET
    @Produces(MediaType.APPLICATION_JSON)
    public List<DataClass> doThings() {
        List<DataClass> list = new ArrayList<>();
        list.add(new DataClass("a", null));
        list.add(new DataClass(null, new InnerDataClass(0, 1)));
        return list;
    }

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    @SneakyThrows
    public String doThingsWithString() {
        List<DataClass> list = new ArrayList<>();
        list.add(new DataClass("a", null));
        list.add(new DataClass(null, new InnerDataClass(0, 1)));
        return new ObjectMapper().writeValueAsString(list);
    }
```

Here's the results:
```bash
$ curl localhost:8080/ -H 'Accept: application/json'   
[{"field1":"a"},{"field1":null,"field2":{"x":0,"y":1}}] 
━━━━━━━━━━━━━━━━━━━━                 
$ curl localhost:8080/ -H 'Accept: text/plain'                                                                       
[{"field1":"a","field2":null},{"field1":null,"field2":{"x":0,"y":1}}] 
```

**Comment by patrox** [comment]
hi @toadzky, I was a bit curious about your issue - so I downloaded your project, took a look and I've a possible suspect - as when I removed the lombok from picture I got the same results from both endpoints:

```
$ curl localhost:8080/ -H 'Accept: text/plain'
```
```
[{"field1":"a","field2":null},{"field1":null,"field2":{"x":0,"y":1}}]
```

```
$ curl localhost:8080/ -H 'Accept: application/json'   
```
```
[{"field1":"a","field2":null},{"field1":null,"field2":{"x":0,"y":1}}]
```

BTW, `@JsonInclude(ALWAYS)` is not needed (as @nickbabcock already wrote), as it's the default one.

**Comment by nickbabcock** [comment]
Haha, I just did the same exact thing that @patrox did and arrived at the same conclusion that it is the combination with lombok that is the problem.

**Comment by patrox** [comment]
@toadzky - I took a step further and found something which might help you, as I imaging that removing lombok for you might be not acceptable:

https://github.com/paradoxical-io/jackson-lombok

Please give it a try and let us know.

**Comment by toadzky** [comment]
I tried it with and without lombok and got inconsistent results, so I didn't mention lombok as a possible cause. In this case, Lombok generates getters correctly, and the manually created object mapper works as expected, but the built in one does not.

As for the link @patrox provided, the project is misnamed IMO because it doesn't have anything to do with Lombok specifically, it has to do with the `@ConstructorProperties` annotation, which is a standard JavaEE annotation. Jackson 2.7+ support `@ConstructorProperties` by default, and is only relevant in PARSING, not serializing.

**Comment by patrox** [comment]
OK, looks like the url which I posted has been a dead-end, but I've been able to pinpoint what exactly is wrong in this scenario.

Seems like there is an interference between Jackson AfterBurner module and Lombok (which is not so surprising, given the fact that both are messing with the Java bytecode).

I've found it out by looking up how the internal `ObjectMapper` used for Jackson in Dropwizard is configured:

https://github.com/dropwizard/dropwizard/blob/master/dropwizard-jackson/src/main/java/io/dropwizard/jackson/Jackson.java#L58

When I tried to serialize a Lombok annotated class using `ObjectMapper` with `AfterburnerModule`, it indeed was skipping the null fields, once I removed the `AfterburnerModule`, the null fields were included.

Besides, I've also found this past issue: https://github.com/dropwizard/dropwizard/issues/1111, describing how the set-up a customized `ObjectMapper`, in case if some of the pre-registered modules are not needed.
